src/control/lib/netdetect/netdetect.go

```diff
@@ -430,6 +432,67 @@ func getNUMASocketID(topology C.hwloc_topology_t, node C.hwloc_obj_t) (uint, err
 	return 0, nil
 }
 
+// NumaAware verifies that NUMA data is available to process
+func NumaAware() (bool, error) {
+	deviceScanCfg, err := initDeviceScan()
+	if err != nil {
+		return false, err
+	}
+	defer cleanUp(deviceScanCfg.topology)
+
+	depth := C.hwloc_get_type_depth(deviceScanCfg.topology, C.HWLOC_OBJ_NUMANODE)
+	numObj := int(C.hwloc_get_nbobjs_by_depth(deviceScanCfg.topology, C.uint(depth)))
+
+	return numObj > 0, nil
+}
```

Comment:
This is a nice solution. I like this a lot because it ought to come in handy for other use cases. We can bypass a lot of code and error checking if we know that there's no point in even going down that code path.

Reply:
Thank you.  Glad you like it.